static/js/message_edit.js

```diff
@@ -156,7 +156,7 @@ exports.save = function (row, from_topic_edited_only) {
 
     var request = {message_id: message.id};
     if (topic_changed) {
-        request.subject = new_topic;
+        util.set_message_topic(request, new_topic);
```

Comment:
This one seems a bit weird; I think we might want another function name (because fundamentally, this is the topic in a `message_edit` API request, not in a frontend-only data structure or the send_message API...)

Reply:
So the rationale for having so many setter/getter functions in the first place is that we haven't fully converted the backend, and particularly in the direction of server -> client, we'll be supporting "subject" for a while (as it's trickier to fix the server without breaking mobile/ZT/etc.).

But in the client -> server direction we already made the backend topic-friendly for all the major endpoints.  So I'd prefer to have a single `set_message_topic` API for anything that's touching message-like structures that are server-bound, particularly since you're often starting with the same data structures.  Does that make sense?

Reply:
I guess my thought is it might make sense to use a different setter function for each endpoint so that one can manage migrations, but I don't feel super strongly.  We can always do such a migration later on.